### ISSUE TITLE
Submiting form is still possible when there is validation error in different tab and some related questions

### ISSUE THREAD

**Issue by kacperdev** [formio/angular#242]
Hi,
I'm trying to integrate that library with my project. But I have few issues with that:

1. So my form have tabs in layout. When you fill form incorrectly in one tab and then move to another (which is filled correctly) form can be submitted. 

2. disableOnInvalid option in submit button doesn't work at all. Are you going to fix it in upcoming releases?

3. Is there any way to suppress number of alerts (or even disable these alerts) which are shown on screen when form is invalid? I'm asking because in my form i'm using array of required fields, so there are many of same alerts.

4. Can I move "Submission Complete" alert to the bottom of form?

5. Invalid pattern message only contains "does not match the pattern". Can I anyhow extend it with label of that field?

Nevertheless I have to admit that this library is amazing and keep it up, because you're doing great work 😃

**Comment by travist** [comment]
Thanks for the compliments. It does seem like you stumbled across some bugs in our renderer so we will add those to our development backlog to get fixed. If you would like to take a stab at fixing these yourself, all of the source code for the "renderer" is found @ https://github.com/formio/formio.js. This library just wraps around that library to render the forms.

**Comment by kacperdev** [comment]
@travist I have encountered another problem. I'd like to pre-populate rendered form with some values. So I've added EventEmitter and binded it to [refresh]. Then I emit some values as submission. It's working well. But when I want to edit any pre-populated data in form, old values are kept in (change) and (submit) event (change event is detected correctly but it doesn't have any affect on event.data).
Is there any solution to this problem? 
I'm using angular-formio 3.11.6 and formiojs 3.20.0 with Angular 7.

Thanks for your help!
